subsys/usb/device_next/usbd_core.c

```diff
@@ -193,7 +193,8 @@ static void usbd_thread(void *p1, void *p2, void *p3)
 		k_msgq_get(&usbd_msgq, &event, K_FOREVER);
 
 		STRUCT_SECTION_FOREACH(usbd_context, uds_ctx) {
-			if (uds_ctx->dev == event.dev) {
+			if (uds_ctx->dev == event.dev &&
+			    usbd_is_initialized(uds_ctx)) {
```

Comment:
Does anything actually prevent two separate context structures from being with the same underlying device? I think this might only be fixing the crash in some specific scenarios (only one context initialized) and not always.

I also don't quite like the fact that `usbd_is_initialized()` check is performed for every single message.

Reply:
> Does anything actually prevent two separate context structures from being with the same underlying device?

No. e.g. Use case is the USB shell's own context.

>I think this might only be fixing the crash in some specific scenarios (only one context initialized) and not always.

Contexts with the same controller are mutually exclusive, only one controller/context can be initialized.

>I also don't quite like the fact that usbd_is_initialized() check is performed for every single message.

Dropped it, I will come up with another solution but not for v3.7.0.